fix(linter): fix cli argument parsing

camc314 opened the pull request:
fixes #14071

Simplify CLI argument collection by unconditionally skipping the first two entries in std::env::args_os(). This ensures correct handling in wrapper scenarios (e.g., Node.js invocations via npm/pnpm) where the conditional "node" check may fail, such as with version managers like asdf. Previously, mis-skipping could lead to wrapper script paths (potentially containing "..") being passed to the CLI parser, triggering validation errors. Since oxlint is not always distributed via the node (no longer a binary), we don't have to worry about the case where we're not running in node.

camc314 commented:
* **#14112** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14112?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14112?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=14112).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on apps/oxlint/src/run.rs:
```diff
@@ -66,13 +66,7 @@ fn lint_impl(load_plugin: JsLoadPluginCb, lint_file: JsLintFileCb) -> CliRunResu
-        args.next();
-    }
-    let args = args.collect::<Vec<_>>();
+    let args = std::env::args_os().skip(2).collect::<Vec<_>>();
```
oxlint isn't distributed as a binary any more so this is fine.

overlookmotel reviewed (approved):
Pushed a commit adding a comment as to why we skip the first 2 args.

Merging. @camc314 please correct the comment if I have it wrong.

overlookmotel commented:
### Merge activity

* **Sep 25, 12:13 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
fix(linter): fix cli argument parsing (#14112)

fixes #14071

Simplify CLI argument collection by unconditionally skipping the first two entries in std::env::args_os(). This ensures correct handling in wrapper scenarios (e.g., Node.js invocations via npm/pnpm) where the conditional "node" check may fail, such as with version managers like asdf. Previously, mis-skipping could lead to wrapper script paths (potentially containing "..") being passed to the CLI parser, triggering validation errors. Since oxlint is not always distributed via the node (no longer a binary), we don't have to worry about the case where we're not running in node.
